docs/components/List/List.stories.mdx

```diff
@@ -1,9 +1,178 @@
 import { Meta } from "@storybook/addon-docs";
+import { Card } from "@jobber/components/Card";
+import { List } from "@jobber/components/List";
 
 <Meta title="Components/Lists and Tables/List" />
 
 # List
 
-Displays a collection of objects that are either the same type or related to one
-another, like billing or activity feed. The main purpose of a list is to help
-the user find an object that leads to its full representation.
+The `List` component renders a collection of items in a structured way. It
+supports both [flat lists](#basic-flat-list) and
+[sectioned lists](#sectioned-list), and allows for
+[custom rendering](#custom-rendering-advanced-usage) of list items via a render
+function.
+
+## Design and Usage Guidelines
+
+The `List` component is a great solution if you are looking to display a
+collection of objects that are either the same type or related to one another,
+like billing or activity feed. The main purpose of a list is to help the user
+find an object that leads to its full representation.
+
+The majority of list implementations can be achieved without needing to
+customize rendering. If your items conform to the ListItemProps interface, the
+component will render them automatically with the default behavior.
+
+#### ListItemProps Interface
+
+When using the default rendering, your items must conform to the
+**`ListItemProps`** interface. Here’s a breakdown of the properties:
+
+**Required:**
+
+- **`id`**: A unique identifier for the list item. This is essential to ensure
+  proper rendering and handling of batch actions.
+
+**Optional:**
+
+- **`title`**: A heading or title for the item.
+
+- **`caption`**: Subdued text shown below the main content.
+
+- **`content`**: Main content, either a string or an array of strings. Supports
+  basic markdown (e.g., **bold** and _italic_).
+
+- **`icon`**: An icon to display on the left side of the content.
+
+- **`iconColor`**: Sets the color of the icon.
+
+- **`isActive`**: Highlights the list item to indicate it needs attention.
+
+- **`section`**: Used to group the list items into sections.
+
+- **`url`**: Makes the item clickable, rendering it as a link.
+
+- **`value`**: A numeric or string value aligned to the right of the item.
+
+- **`onClick`**: A callback function triggered when the item is clicked.
```

Comment:
not sure how I feel about writing out all the props in our docs. we have all this documented in the JSDocs already, it's just that our tooling doesn't expose it in Storybook

![image](https://github.com/user-attachments/assets/888b688d-dbd7-4594-9b30-7348eba7ee03)

on one hand, it is nice to have it available in storybook, on the other hand it's adding something we have to make sure to keep in sync going forward. additionally it means there are 3 places to find props of a component
- component docs
- component examples page
- JS Docs

I'm tempted to say we should avoid duplicating documentation and instead find a solution to nested props to leverage the JSDocs (in general , definitely not in this ticket!)

Reply:
You make a good point, I was being a bit overzealous but it does add another layer to the maintainability so I will remove them.